packages/terser/src/module.ts

```diff
@@ -1,12 +1,31 @@
+import { fileURLToPath } from 'url';
+
 import type { NormalizedOutputOptions, RenderedChunk } from 'rollup';
 import { hasOwnProperty, isObject, merge } from 'smob';
 
 import type { Options } from './type';
 import { WorkerPool } from './worker-pool';
 
 export default function terser(options: Options = {}) {
+  let filePath: string | undefined;
+
+  if (typeof __filename !== 'undefined') {
+    filePath = __filename;
+  } else {
+    // @ts-ignore
+    // eslint-disable-next-line no-lonely-if
+    if (typeof import.meta !== 'undefined') {
```

Comment:
Could be an `else if`?

Reply:
In general of cause, but i couldn't provide `@ts-ignore` to the `else if` statement :/

Reply:
in general we shouldn't have to use `// @ts-ignore`. I can take a look at your fork later tonight or tomorrow to find a better solution.

Reply:
That would be great ☺️. The issue here is this that tsc is complaining about the usage of imoort.meta because the package is not an esm module.

Reply:
Node throws a syntax error for `import.meta` in CJS modules.

Reply:
It is very picky on mixing CJS and ESModule code, it will complain one way or the other and just refuse to run. One solution I found is to have two completely different entrypoint files, one for CJS and one for ESModule. Compile them separately, then in `package.json`, define `main` for CJS and `module` for ESModule.

Reply:
does it complain about using `global['import']`?

Reply:
@shellscape `global['import']` do not complain in CJS mode, but it does not work in MJS mode either, because `import` is a special keyword, that `global['import'].meta.url` is different than `import.meta.url`.

Having the following code:

``` js
console.log(global['import'])
```

Run it as `node test.cjs`:
```
undefined
```
Run it as `node test.mjs`:
```
undefined
```

Reply:
You could also introduce an isomorphic `__filename` as suggested in [this blog post](https://antfu.me/posts/isomorphic-dirname):

```ts
import { fileURLToPath } from 'url'

const filePath = typeof __filename !== 'undefined'
    ? __filename
    : fileURLToPath(import.meta.url)
```

Reply:
@mxdvl That article is outdated. It won’t work anymore in latest node because keyword import detected in anywhere in CSJ mode is now a syntax error.

Reply:
Oh, could [`@rollup/plugin-replace`](https://www.npmjs.com/package/@rollup/plugin-replace) be an alternative? That might be lighter touch than two separate entry points.

Reply:
that was my initial thought but _would_ double the size of the package

Reply:
> @mxdvl That article is outdated. It won’t work anymore in latest node because keyword import detected in anywhere in CJS mode is now a syntax error.

Besides, that was exactly what i have implemented in the pr. ^^

Reply:
> that was my initial thought but _would_ double the size of the package

but wouldn't it just affect the size during development/build, because it would be a defined as dev dependency ?